src/Raven.Server/Documents/Handlers/BatchHandler.cs

```diff
@@ -21,54 +26,76 @@ public class BatchHandler : DatabaseRequestHandler
         [RavenAction("/databases/*/bulk_docs", "POST")]
         public async Task BulkDocs()
         {
-            DocumentsOperationContext ctx;
-            using (ContextPool.AllocateOperationContext(out ctx))
+            using (ContextPool.AllocateOperationContext(out DocumentsOperationContext ctx))
+            using (var mergedCmd = new MergedBatchCommand{Database = Database})
             {
-                var cmds = await BatchRequestParser.ParseAsync(ctx, RequestBodyStream(), Database.Patcher);
-               
-                var waitForIndexesTimeout = GetTimeSpanQueryString("waitForIndexesTimeout", required: false);
-
-                using (var mergedCmd = new MergedBatchCommand
+                if (HttpContext.Request.ContentType != null) // multipart/mixed
```

Comment:
We are assuming this is rare, and as such, it need to go to its own method, so the common code will be smaller.